### PR TITLE
Add repo revision option

### PR DESCRIPTION
**Pull request by petrukha-ivan**
### What 
Added `revision` parameter to specify a branch, tag, or commit when downloading a model.

### CODE REVIEW

File: Sources/Hub/HubApi.swift [hunk 1]
```diff
@@ -349,7 +350,7 @@ public extension HubApi {
                 url = url.appending(component: repo.type.rawValue)
             }
             url = url.appending(path: repo.id)
-            url = url.appending(path: "resolve/main") // TODO: revisions
+            url = url.appending(path: "resolve/\(revision)")
```
Comment (FL33TW00D) [inline Sources/Hub/HubApi.swift:353]: Maybe best not to do direct string interpolation?
Also needs a null check I think

Comment (petrukha-ivan) [reply Sources/Hub/HubApi.swift:353]: String interpolation is the preferred method for this in Swift because it's faster and benefits from compile-time optimizations. We also don't need a null check here since revision is not marked as `Optional` and always has a value. However, I can add something like `assert(revision.isEmpty == false, "Revision cannot be empty")` if you'd prefer 🙂

**Comment by FL33TW00D** [comment]
Just tested:
```
~/Code/swift-transformers test-revision ❯ .build/release/hub-cli download jinaai/jina-embeddings-v3 --revision BINGO 
Error: Authentication required. Please provide a valid Hugging Face token.
~/Code/swift-transformers test-revision ❯ .build/release/hub-cli download jinaai/jina-embeddings-v3 --revision main 
40.00%
~/Code/swift-transformers test-revision ❯ .build/release/hub-cli download jinaai/jina-embeddings-v3 --revision INVALID$%£@ 
Error: Authentication required. Please provide a valid Hugging Face token.
```
Works fine with valid revision, incorrect error message if not. Would add a test with janky characters too, can never be too sure!

**Comment by petrukha-ivan** [comment]
@FL33TW00D Hi! You're right 👍 I've fixed the error type to throw the correct error in case of invalid revision, and also added a few unit tests to test both positive and negative cases.

**Review by pcuenca — APPROVED** [review]
Looks good to me, thanks for the great contribution and for iterating 🙌 !

I think we should maybe merge #200 first, just in case. Would you agree, @FL33TW00D?

**Comment by FL33TW00D** [comment]
Still seeing an issue here @petrukha-ivan, try the following:
`.build/release/hub-cli download jinaai/jina-embeddings-v3 --revision main`
`.build/release/hub-cli download jinaai/jina-embeddings-v3 --revision refactor-task-type`

**Comment by petrukha-ivan** [comment]
@FL33TW00D Ah, good catch — it's fixed now 👍

**Comment by pcuenca** [comment]
One last thing: `getFileMetadata` is still using `main`, I think. Can we pass revision through to it?

**Comment by FL33TW00D** [comment]
@petrukha-ivan Nothing you've done, we are getting 429'd by the Hub

**Comment by FL33TW00D** [comment]
@petrukha-ivan Appreciate the back and forth here! Thanks for the contribution 🤗
